modules/styleguide/module.php

```diff
@@ -0,0 +1,150 @@
+		return 'styleguide';
+	}
+
+	public static function is_active() {
+		return Plugin::$instance->experiments->is_feature_active( self::EXPERIMENT_NAME );
+	}
+
+	public static function get_experimental_data() {
+		return [
+			'name' => static::EXPERIMENT_NAME,
+			'title' => esc_html__( 'Global Style Guide', 'elementor' ),
+			'tag' => esc_html__( 'Feature', 'elementor' ), //todo : add tag
+			'description' => esc_html__( 'Display a live preview of changes to global colors and fonts in a sleek style guide from the site’s settings. You will be able to toggle between the style guide and the page to see your changes in action.', 'elementor' ),
+			'release_status' => Experiments_Manager::RELEASE_STATUS_ALPHA,
+			'default' => Experiments_Manager::STATE_INACTIVE,
+		];
+	}
+
+	/**
+	 * Enqueue scripts.
+	 *
+	 * @return void
+	 */
+	public function enqueue_main_scripts() {
+		wp_enqueue_script(
+			$this::ASSETS_HANDLE,
+			$this->get_js_assets_url( $this::ASSETS_SRC ),
+			[ 'elementor-editor' ],
+			ELEMENTOR_VERSION,
+			true
+		);
+
+		$kit_id = Plugin::$instance->kits_manager->get_active_id();
+
+		wp_localize_script( $this::ASSETS_HANDLE, 'elementorStyleguideConfig', [
+			'activeKitId' => $kit_id,
+		] );
+	}
+
+	public function enqueue_app_initiator( $is_preview = false ) {
+		$dependencies = [
+			'react',
+			'react-dom',
+		];
+
+		if ( ! $is_preview ) {
+			$dependencies[] = $this::ASSETS_HANDLE;
+		}
+
+		wp_enqueue_script(
+			$this::ASSETS_HANDLE . '-app-initiator',
+			$this->get_js_assets_url( $this::ASSETS_SRC . '-app-initiator' ),
+			$dependencies,
+			ELEMENTOR_VERSION,
+			true
+		);
+	}
+
+	public function enqueue_styles() {
+		wp_enqueue_style(
+			$this::ASSETS_HANDLE,
+			$this->get_css_assets_url( 'modules/' . $this::ASSETS_SRC . '/' . $this::ASSETS_SRC ),
+			[],
+			ELEMENTOR_VERSION
+		);
+	}
+
+	public function register_controls() {
+		$controls_manager = Plugin::$instance->controls_manager;
+
+		$controls_manager->register( new Switcher() );
+	}
+
+	/**
+	 * Add the Enable Styleguide Preview controls to Global Colors and Global Fonts.
+	 *
+	 * @param $element
+	 * @param string $section_id
+	 * @param array $args
+	 */
+	public function add_styleguide_enable_controls( $element, $section_id, $args ) {
+		if ( 'kit' !== $element->get_name() || ! in_array( $section_id, [ 'section_global_colors', 'section_text_style' ] ) ) {
+			return;
+		}
+
+		$control_name = str_replace( 'global-', '', $args['tab'] ) . '_enable_styleguide_preview';
+
+		$element->add_control(
+			$control_name,
+			[
+				'label' => esc_html__( 'Style Guide Preview', 'elementor' ),
+				'type' => Switcher::CONTROL_TYPE,
+				'description' => esc_html__( 'Switch between the content area and style guide to preview your changes to global colors.', 'elementor' ),
+				'separator' => 'after',
+				'label_off' => esc_html__( 'Off', 'elementor' ),
+				'label_on' => esc_html__( 'On', 'elementor' ),
+				'on_change_command' => true,
+			]
+		);
```

Comment:
Consider making it frontend only, and reading from the user preferences. + move here the code that was added in the user preferences file.

Reply:
Instead. I added `this.model.set( 'return_value', null );` to switcher.js so now it doesn't really save the value. This allowed me to remove the exclusion from site settings export process.

Together with that, I didn't want to add the on_change to the base, and as I need it to behave in a custom way upon change, I'm leaving the custom control for now.